teuthology/packaging.py

```diff
@@ -973,12 +973,40 @@ def _get_repo(self):
         return resp.text
 
     def _install_rpm_repo(self):
+        dist_release = self.dist_release
         repo = self._get_repo()
-        sudo_write_file(
-            self.remote,
-            '/etc/yum.repos.d/{proj}.repo'.format(proj=self.project),
-            repo,
-        )
+        if dist_release in ['opensuse', 'sle']:
+            #
+            # Shaman does not currently return opensuse repos in a format that zypper
+            # understands -- see https://tracker.ceph.com/issues/44183 for details.
+            #
+            # So, do some text manipulation to convert the yum-style repo to
+            # zypper-style.
+            #
+            # This text manipulation/conversion code should continue to work even
+            # after Shaman is fixed.
+            #
+            munged_repo = ''
+            repo_lines = repo.splitlines()
+            for repo_line in repo_lines:
+                if repo_line.startswith('baseurl='):
+                    if repo_line.endswith('$basearch'):
```

Comment:
I wonder if you can reuse method `rsstrip` from `teuthology.repo_utils`. Ideally to move it to teuthology.utils of course.

Reply:
This code is only a temporary workaround. It will be removed once https://tracker.ceph.com/issues/44183 is fixed.

Reply:
> This text manipulation/conversion code should continue to work even
             # after Shaman is fixed

Means no-one will hurry to remove this code, as practice shows people start copy-pasting what ever code in the new patches, and they usually say they don't want to fix because they just copied from other place of codebase, so they may use this as an example. If it is hard to do, then ok